File: leetcode-js/071-simplify-path.js
/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function(path) {
    'use strict';

    const parts = path.split('/');
    const stack = [];

    for (const part of parts) {
        switch (part) {
        case '':
        case '.':
            break;
        case '..':
            stack.pop();
            break;
        default:
            stack.push(part);
            break;
        }
    }

    return '/' + stack.join('/');
};

File: leetcode-java/071-simplify-path.java
public class Solution {
    public String simplifyPath(String path) {
        Stack<String> stack = new Stack<>();

        for (String part : path.split("/")) {
            if (part.equals("..")) {
                if (!stack.empty()) {
                    stack.pop();
                }
            } else if (!part.equals("") && !part.equals(".")) {
                stack.push(part);
            }
        }

        if (stack.empty()) {
            return "/";
        } else {
            StringBuilder sb = new StringBuilder();
            for (String part : stack) {
                sb.append("/" + part);
            }
            return sb.toString();
        }
    }
}
